common.tf

```diff
@@ -13,15 +13,15 @@ locals {
 data "aws_caller_identity" "current" {}
 
 data "aws_vpc" "waggledance_vpc" {
-  id = "${var.vpc_id}"
+  id = var.vpc_id
 }
 
 data "aws_secretsmanager_secret" "bastion_ssh_key" {
   count = "${var.bastion_ssh_key_secret_name == "" ? 0 : 1}"
```

Comment:
I'm confused as to why some variable references throughout this PR are still using the `${var.name}" convention.  If that method of referencing vars still works in Terraform 0.12, why did you need to change the majority of them?

Reply:
new code is not backward compatible because of this line, main reason for this PR
https://github.com/ExpediaGroup/apiary-federation/blob/c7f6b7b0b628d6e8781eff6a68622f4bde4db769/endpoints.tf#L49

since it is not backward compatible anyway, updated most code to use recommended way of using variables in TF 0.12, main change in TF 0.12 is to support data types other than strings for parameters and variables, so it is not needed to enclose everything in `"${......}"`

Reply:
The OCD in me hates that some of the variable references are still in old, pre-0.12 format.  Unless there is a compelling reason to keep them, can you change them all?  Otherwise, it is just confusing to people who haven’t dealt with different versions of Terraform.

Reply:
sounds good, I will have another pass.
in general, we can use new format for most variables, might need to use old format when we are construction string like `"prefix-${var.variable}-suffix"`